### DISCUSSION TITLE
"zsh: #: command not found..." error on startup

### DISCUSSION THREAD

**Discussion by arielcorte** [ohmyzsh/ohmyzsh#11862] [Help] [answered]
Every time I open a terminal, or create a new pane in tmux it shows

```
zsh: ﻿#: command not found...
➜  ~ 
```

Then I can use the terminal normally, but it's annoying viewing that error every time. How can I get rid of it?

I reseted .zshrc. Inspected .oh-my-zsh.sh command. Checked the $ZSH path. Reseted i3 setup and even commented parts of it and checked if the dependencies it run where installed. Reseted .bashrc.
Nothing of all those worked, maybe I'm missing something.

**Comment by carlosala** [comment]
Which is the full line? For example, this would be an example `zsh: command not found: doesnotexist`

**Comment by arielcorte** [comment]
@carlosala That's the fun part, there's no more, it literally just says that:
![image](https://github.com/ohmyzsh/ohmyzsh/assets/86988649/3bf3d97b-ac9c-4166-a170-9f5af66768d2)

**Reply by carlosala** [reply]
I could try to debug it. Could you send your initialization logs to logs@ohmyz.sh? You can see how to get them with the last snippet [here](https://github.com/ohmyzsh/ohmyzsh/wiki/Troubleshooting#other-problems).

**Reply by mcornella** [reply]
The error shows the failed command at the start, so you should look for a line in your .zshrc or .zshenv files starting with `#:`.

**Reply by arielcorte** [reply]
@carlosala Sent!

**Reply by arielcorte** [reply]
@mcornella I've checked my .zshrc and .zshenv files and didn't find any line starting with `#:`, just the comments.

**Reply by mcornella** [reply]
It looks like the part that fails is the first line in `.zshrc`:

```sh
﻿# If you come from bash you might have to change your $PATH.
+/home/user/.zshrc:1> '﻿#' If you come from bash you might have to change your /home/user/.cargo/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/var/lib/snapd/snap/bin.
+command_not_found_handle:1> local runcnf=1
+command_not_found_handle:2> local retval=127
+command_not_found_handle:5> [[ 569Xivx == *i* ]]
+command_not_found_handle:8> [[ ! -S /run/dbus/system_bus_socket ]]
+command_not_found_handle:11> [[ ! -x /usr/libexec/packagekitd ]]
+command_not_found_handle:14> [[ -n '' ]]
+command_not_found_handle:17> [[ ! -x /usr/libexec/pk-command-not-found ]]
+command_not_found_handle:20> [ 1 -eq 1 ']'
+command_not_found_handle:21> /usr/libexec/pk-command-not-found '﻿#' If you come from bash you might have to change your /home/user/.cargo/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/var/lib/snapd/snap/bin.

zsh: ﻿#: command not found...
```

It is interpreting the `#` as a command, and not as a comment. This is not happening with the rest of comments, so my guess is that there is a character before or after making it different than a regular # character.

You can use `hexdump -C .zshrc | head -5` to confirm the binary code of the first 5 lines. Either way, the solution might be just to delete the first lines and recreate them.

**Comment by arielcorte** [comment, marked as answer]
I fixed it!
In my zsh logs i found the next lines:

```bash
<feff># If you come from bash you might have to change your $PATH.
+/home/arielcorte/.zshrc:1> '<feff>#' If you come from bash you might have to change your /home/arielcorte/.cargo/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/var/lib/snapd/snap/bin.
+command_not_found_handle:1> local runcnf=1
+command_not_found_handle:2> local retval=127
+command_not_found_handle:5> [[ 569Xivx == *i* ]]
+command_not_found_handle:8> [[ ! -S /run/dbus/system_bus_socket ]]
+command_not_found_handle:11> [[ ! -x /usr/libexec/packagekitd ]]
+command_not_found_handle:14> [[ -n '' ]]
+command_not_found_handle:17> [[ ! -x /usr/libexec/pk-command-not-found ]]
+command_not_found_handle:20> [ 1 -eq 1 ']'
+command_not_found_handle:21> /usr/libexec/pk-command-not-found '<feff>#' If you come from bash you might have to change your /home/arielcorte/.cargo/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/var/lib/snapd/snap/bin.
zsh: <feff>#: command not found...
+command_not_found_handle:22> retval=127 
+command_not_found_handle:28> return 127
# export PATH=$HOME/bin:/usr/local/bin:$PATH
```
Note: I have manually added the \<feff> in my comment, but in my file there was the actual character.

then I went to both my .bashrc and .zshrc files and with vim I used `:set nobomb` (actually it only changed the .zshrc file, but I tested both just in case) and it was fixed :)

Thanks @carlosala for the log snippet and help and @mcornella for the hint for looking for `#:` (I find it in my logs).
